library/modules/bigip_device_dns.py

```diff
@@ -104,153 +105,252 @@
-        HAS_F5SDK = False
+    from ansible.module_utils.network.f5.common import exit_json
+    from ansible.module_utils.network.f5.common import fail_json
+    from ansible.module_utils.network.f5.common import is_empty_list
 
 
 class Parameters(AnsibleF5Parameters):
     api_map = {
-        'dhclient.mgmt': 'dhcp',
         'dns.cache': 'cache',
         'nameServers': 'name_servers',
-        'include': 'ip_version'
+        'include': 'ip_version',
     }
 
     api_attributes = [
-        'nameServers', 'search', 'include'
+        'nameServers', 'search', 'include',
     ]
 
     updatables = [
-        'cache', 'name_servers', 'search', 'ip_version'
+        'cache', 'name_servers', 'search', 'ip_version',
     ]
 
     returnables = [
-        'cache', 'name_servers', 'search', 'ip_version'
+        'cache', 'name_servers', 'search', 'ip_version',
     ]
 
     absentables = [
-        'name_servers', 'search'
+        'name_servers', 'search',
     ]
 
-    def to_return(self):
-        result = {}
-        for returnable in self.returnables:
-            result[returnable] = getattr(self, returnable)
-        result = self._filter_params(result)
-        return result
 
+class ApiParameters(Parameters):
+    pass
+
+
+class ModuleParameters(Parameters):
     @property
     def search(self):
-        result = []
-        if self._values['search'] is None:
+        search = self._values['search']
+        if search is None:
             return None
-        for server in self._values['search']:
-            result.append(str(server))
-        return result
+        if isinstance(search, str) and search != "":
+            result = list()
+            result.append(str(search))
+            return result
+        if is_empty_list(search):
+            return []
+        return search
 
     @property
     def name_servers(self):
-        result = []
-        if self._values['name_servers'] is None:
+        name_servers = self._values['name_servers']
+        if name_servers is None:
             return None
-        for server in self._values['name_servers']:
-            result.append(str(server))
-        return result
+        if isinstance(name_servers, str) and name_servers != "":
+            result = list()
+            result.append(str(name_servers))
+            return result
+        if is_empty_list(name_servers):
+            return []
+        return name_servers
 
     @property
     def cache(self):
+        if self._values['cache'] is None:
+            return None
         if str(self._values['cache']) in ['enabled', 'enable']:
             return 'enable'
         else:
             return 'disable'
 
-    @property
-    def dhcp(self):
-        valid = ['enable', 'enabled']
-        return True if self._values['dhcp'] in valid else False
-
     @property
     def ip_version(self):
-        if self._values['ip_version'] in [6, '6', 'options inet6']:
+        if self._values['ip_version'] == 6:
```

Comment:
This change here drops the normalizing of certain API params. I dont think this is intentional?

Reply:
i have never seen the other params, this is correct, the only thing bigip returns is options inet6, so that was redundant